plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/glue/InMemoryGlueCache.java

```diff
@@ -81,21 +92,43 @@ private record FunctionKey(String databaseName, String functionName) {}
     private final AtomicLong tableInvalidationCounter = new AtomicLong();
     private final AtomicLong partitionInvalidationCounter = new AtomicLong();
 
-    public InMemoryGlueCache(Duration metadataCacheTtl, Duration statsCacheTtl, long maximumSize)
+    public InMemoryGlueCache(
+            CatalogName catalogName,
+            Duration metadataCacheTtl,
+            Duration statsCacheTtl,
+            Optional<Duration> refreshInterval,
+            int maxMetastoreRefreshThreads,
+            long maximumSize)
     {
+        if (refreshInterval.isPresent()) {
+            refreshExecutor = Optional.of(newCachedThreadPool(daemonThreadsNamed("hive-metastore-" + catalogName + "-%s")));
```

Comment:
Why do you need Optional. If thread pool is not used it should not populate threads.